probe_accuracy_test_suite.py

```diff
@@ -15,540 +15,806 @@
+KLIPPY_LOG = "~/klipper_logs/klippy.log"
+DATA_DIR = "/tmp"
 RUNID = datetime.now().strftime("%Y%m%d_%H%M")
-CFG = {}
-TOOLHEAD = {}
-isKlicky = False
-isKlippain = False
-isTap = False
-safe_z = None
+CLEAR_LINE = "\033[1A\x1b[2K"
 
+class Probe():
+    def __init__(self, printer):
+        self.printer = printer
 
-def main(userparams):
-    if not os.path.exists(DATA_DIR):
-        os.mkdir(DATA_DIR)
-    
-    if userparams["detect_probe"]:
-        CFG.update(query_printer_objects("configfile", "config"))
-        TOOLHEAD.update(query_printer_objects("toolhead"))
+        self.isKlicky = False
+        self.isKlippain = False
+        self.isTap = False
+
+        self._detect()
+
+    def is_present(self):
+        return (
+                self.isKlicky
+            or  self.isKlippain
+            or  self.isTap
+        )
+
+    def lock(self, lock = True):
+        if self.isKlicky:
+            self.printer.gcode("ATTACH_PROBE_LOCK")
+        if self.isKlippain:
+            if lock:
+                self.printer.gcode("ACTIVATE_PROBE LOCK=true")
+            else:
+                self.printer.gcode("ACTIVATE_PROBE")
+
+    def unlock(self, unlock = False):
+        if self.isKlicky:
+            self.printer.gcode("DOCK_PROBE_UNLOCK")
+        elif self.isKlippain:
+            if unlock:
+                self.printer.gcode("DEACTIVATE_PROBE UNLOCK=true")
+            else:
+                self.printer.gcode("DEACTIVATE_PROBE")
+
+    def check_error(msg):
+        klicky_macro_issue = " ".join([
+            "!! Error evaluating 'gcode_macro PROBE_ACCURACY:gcode':",
+            "CommandError:",
+            "Must perform PROBE_ACCURACY with the probe above the BED!"
+        ])
+        if klicky_macro_issue == msgs:
+            print("This issue can be fixed by updating klicky-macros.cfg")
+            print(
+                "Reference: https://github.com/jlas1/Klicky-Probe/commit/31a481c843567233c807bb310b6f0e83d60b4fca"
+            )
+            return
+
+        print("Unknown probe error.")
+        print("Exiting!")
+        sys.exit(1)
     
-        detect_probe()
-    else:
+    def _detect(self):
+        print("Probe type: ..." )
+
         try:
-            CFG.update(query_printer_objects("configfile", "config"))
-            TOOLHEAD.update(query_printer_objects("toolhead"))
-            detect_probe()
-            
-            homing()
-            move_to_safe_z()
-
-            if not any(
-                [
-                    userparams["corner"]
-                    or userparams["repeatability"]
-                    or userparams["drift"]
-                    or userparams["speedtest"]
-                ]
-            ):
-                print("Running all tests")
-                userparams.update({"corner": 30, "repeatability": 20, "drift": 100})
+            settings = self.printer.query("configfile", "settings")
+            if settings["docklocation_x"]:
```

Comment:
this doesn't cover the case of using the [dockable_probe] module, although there are starting to be so many versions of the docking macros that makes writing something like this a pain, as there might not be a good way to cover all edge cases elegantly. 

```[dockable_probe]
pin:
z_offset:
sample_retract_dist:
approach_position:
dock_position:
detach_position:
(check_open_attach: OR probe_sense_pin:) AND/OR dock_sense_pin:
```